lib/spack/spack/fetch_strategy.py

```diff
@@ -907,51 +926,51 @@ def __init__(self, root):
         self.root = os.path.abspath(root)
 
     def store(self, fetcher, relativeDst):
+        # skip fetchers that aren't cachable
         if not fetcher.cachable:
             return
 
+        # Don't store things that are already cached.
+        if isinstance(fetcher, CacheURLFetchStrategy):
```

Comment:
I'm worried that if cachable returns false unconditionally, it'll one day be misinterpreted by the mirror logic to mean "don't mirror this".  Things that aren't strictly "cachable" should still be mirrored so that you can build them across the air gap.  So I tried to keep that logic confined to `FsCache`, which is the special case... what do you think?